R/init.R

```diff
@@ -205,13 +207,30 @@ init <- function(data,
     }
   }
 
+  # In case of a "Landing Popup", do not create a module for it. Just extract the module and use directly in server.
+  # Assuming "Landing Popup" is not used in a nested module.
+  labels <- module_labels(modules)
+  lp_cond <- "Landing_Popup" %in% names(labels)
+
+  if (!lp_cond && "Landing Page" %in% unlist(labels)) {
+    stop("Please do not use a module labelled 'Landing Popup' within a nested module.")
+  } else if (lp_cond) {
+    landing_popup <- modules$children[which(lp_cond)]
+    modules$children <- modules$children[-which(lp_cond)]
+  } else {
+    landing_popup <- NULL
+  }
+
   # Note regarding case `id = character(0)`:
   # rather than using `callModule` and creating a submodule of this module, we directly modify
   # the `ui` and `server` with `id = character(0)` and calling the server function directly
   # rather than through `callModule`
   res <- list(
     ui = ui_teal_with_splash(id = id, data = data, title = title, header = header, footer = footer),
     server = function(input, output, session) {
+      if (length(landing_popup) > 0L) {
+        do.call(landing_popup[[1]]$server, c(list(id = "landing_module_shiny_id"), landing_popup[[1]]$server_args))
```

Comment:
```suggestion
        if (length(landing_popup) > 1L) {
          message("There are more than one Landing Popup module. teal will only execute the first one.")
        }
        do.call(landing_popup[[1]]$server, c(list(id = "landing_module_shiny_id"), landing_popup[[1]]$server_args))
```

Informing user that in the case of multiple popup landing, we will only execute the first one.

Reply:
thanks @donyunardi or we can execute all of them honestly, one above another. If someone would like to have 2 popups that would allow him/her to do that. This is pretty cool idea.

Reply:
```
message("There are more than one Landing Popup module. teal will only execute the first one.")
```
I don't think that would be relevant information to the app user, rather to the developer. This was a draft so I didn't consider the app dev putting more than one in.

I think we should take more time to define the purpose of this feature ant this special class of module.

Shiny cannot display two modals at once unless you use [tricks](https://github.com/insightsengineering/teal.widgets/pull/199).